### PR TITLE
ci: standardize dockerfiles

### PR DESCRIPTION
**Pull request by ernestojeda**
- [x] use alpine 3.12
- [x] common `apk add` where possible
- [x] update copyright header where possible

Signed-off-by: Ernesto Ojeda <ernesto.ojeda@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: Dockerfile.build [hunk 1]
```diff
@@ -24,7 +24,7 @@ RUN apk add --update make git curl bash zeromq-dev libsodium-dev pkgconfig build
 WORKDIR /edgex-go
 
 # This will be needed to run `make raml_verify`
-COPY --from=docker:latest /usr/local/bin/docker /usr/local/bin/docker
+COPY --from=docker:20.10.0 /usr/local/bin/docker /usr/local/bin/docker
```
[the remark this answers is not in the document]
Comment (ernestojeda) [reply Dockerfile.build:27]: nice catch. I will remove docker from this image.

**Review by lenny-intel — APPROVED** [review]

**Comment by lenny-intel** [comment]
recheck

File: Dockerfile.build [hunk 2]
```diff
@@ -23,8 +23,5 @@ RUN apk add --update make git curl bash zeromq-dev libsodium-dev pkgconfig build
 
 WORKDIR /edgex-go
 
-# This will be needed to run `make raml_verify`
```
Comment (cloudxxx8) [inline Dockerfile.build:26]: should we put --no-cahce in the `RUN apk add` command?

File: cmd/core-data/Dockerfile [hunk 3]
```diff
@@ -40,7 +50,7 @@ EXPOSE $APP_PORT
 # So we can try these.
 RUN sed -e 's/dl-cdn[.]alpinelinux.org/nl.alpinelinux.org/g' -i~ /etc/apk/repositories
 
-RUN apk --no-cache add zeromq
+RUN apk add --no-cache zeromq
```
Comment (cloudxxx8) [inline cmd/core-data/Dockerfile:53]: should we put --update in the `RUN apk add` command?

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**